core/tests/unit/ldml/test_transforms.cpp

```diff
@@ -34,33 +35,53 @@
   }
 #endif
 
-
 // needed for streaming operators
 #include "utfcodec.hpp"
 
 using namespace km::core::ldml;
 using namespace km::core::kmx;
 
-std::u32string marker_list_to_string(const marker_list &m) {
+void
+prepend_hex_oct(std::u32string &str, char32_t x) {
+  for (auto i = 0; i < 8; i++) {
+    KMX_DWORD remainder = x & 0xF;  // get the last nibble
+    char32_t ch;
+    if (remainder < 0xA) {
+      ch = U'0' + remainder;
+    } else {
+      ch = U'A' + (remainder - 0xA);
+    }
```

Comment:
```suggestion
    const char32_t ch = remainder < 0xA ? U'0' + remainder : U'A' + (remainder - 0xA);
```

Not important but shorter